prow/gcsupload/run.go

```diff
@@ -77,7 +77,9 @@ func (o Options) assembleTargets(spec *downwardapi.JobSpec, extra map[string]gcs
 
 	if latestBuilds := gcs.LatestBuildForSpec(spec, builder); len(latestBuilds) > 0 {
 		for _, latestBuild := range latestBuilds {
-			uploadTargets[latestBuild] = gcs.DataUpload(strings.NewReader(spec.BuildID))
+			dir, filename := path.Split(latestBuild)
+			filename, metadata := gcs.MetadataFromFileName(filename)
+			uploadTargets[path.Join(dir, filename)] = gcs.DataUploadWithMetadata(strings.NewReader(spec.BuildID), metadata)
```

Comment:
This is a bit tricky to read since on the surface you're `path.Join` the output of `path.Split` so perhaps let's assign the output of `MetadataFromFileName` to a new variable for clarity

Reply:
> ... so perhaps let's assign the output of `MetadataFromFileName` to a new variable for clarity

Fixed with e76f268 -> 0b7effa.